lib/checkunusedvar.cpp

```diff
@@ -1534,7 +1534,43 @@ void CheckUnusedVar::checkStructMemberUsage()
         if (bailout)
             continue;
 
+        // check if class/struct has default equality operator
+        bool has_default_eq = true;
+        for (const Function &f : scope.functionList) {
+            if (f.isOperator() && f.name() == "operator==") {
+                if (f.isDefault()) {
+                    has_default_eq = true;
+                    break;
+                }
+                has_default_eq = false;
+                break;
+            }
+        }
+
+        // check if default equality is used
+        bool default_eq_used = false;
+        if (has_default_eq) {
+            for (const Token *tok = mTokenizer->tokens(); tok; tok = tok->next()) {
+                if (Token::simpleMatch(tok, "==")) {
+                    const Token *lhs = tok->astOperand1();
```

Comment:
I have the feeling we should double check that some binary operator always have two operands. We don't always catch all syntax errors.